qemu/tests/cpu_info_check.py

```diff
@@ -0,0 +1,137 @@
+import re
+import logging
+
+from avocado.utils import process
+
+from virttest import cpu
+from virttest import env_process
+from virttest import error_context
+from virttest import utils_misc
+
+from virttest.utils_version import VersionInterval
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    cpuinfo query test:
+    1). run query cmd. e.g -cpu ?cpuid
+    2). check the expected info is included in the cmd output.
+    3). Boot guest and check the output of qmp command "qom-list-types"
+    4). Check the output of qmp command "query-cpu-definitions"
+    5). Check the output of qmp command "query-cpu-model-expansion"
+    """
+
+    def remove_models(model_list):
+        """
+        Remove models from cpu_types
+        :param model_list: The list of models to be removed
+        """
+        for model in model_list:
+            try:
+                cpu_types.remove(model)
+            except ValueError:
+                logging.warning('The model to be removed is not'
+                                ' in the list: %s' % model)
+                continue
+
+    def get_patterns(p_list):
+        """
+        Return all possible patterns for given flags
+        :param p_list: The list of flags
+        """
+        r_list = []
+        replace_char = [('_', ''), ('_', '-'), ('.', '-'),
+                        ('.', ''), ('.', '_')]
+        for p in p_list:
+            r_list.extend(list(map(lambda x: p.replace(*x), replace_char)))
+        return set(r_list)
+
+    cpu_types = []
+    list(map(cpu_types.extend, list(cpu.CPU_TYPES.values())))
+
+    qemu_path = utils_misc.get_qemu_binary(params)
+    qemu_version = env_process._get_qemu_version(qemu_path)
+    match = re.search(r'[0-9]+\.[0-9]+\.[0-9]+(\-[0-9]+)?',
+                      qemu_version)
+    host_qemu = match.group(0)
+    remove_models(params.objects('remove_list'))
+    if host_qemu in VersionInterval('[,4.2.0)'):
+        remove_models(params.objects('cpu_model_8'))
+    if host_qemu in VersionInterval('[,3.1.0)'):
+        remove_models(params.objects('cpu_model_3_1_0'))
```

Comment:
@nanliu-r Host rhel7 will go through Line 58 -> Line 60 -> Line 62(since it definitely meets the condition `host_qemu in VersionInterval('[,3.1.0)')`), so all desired removing model will be removed.

Reply:
yes, you are right, seems it is the format issue that causes the case to fail, please help to check.